scripts/tools/EXT_Exp_MetricProgrammable.rst

```diff
@@ -0,0 +1,120 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+    t=tags['$t']
+    T=t.upper()
+%>
+:orphan:
+
+.. _ZET_experimental_programmable_metric:
+
+==========================================
+Metric Programmable Experimental Extension
+==========================================
+
+API
+----
+* Enumerations
+
+    * ${t}_metric_programmable_param_type_exp_t
+
+* Structures
+
+    * ${t}_metric_programmable_exp_properties_t
+    * ${t}_metric_programmable_param_exp_desc_t
+    * ${t}_metric_programmable_param_value_exp_t
+    * ${t}_metric_programmable_param_value_exp_desc_t
+    * ${t}_value_exp_desc_t
+    * ${t}_value_uint64_range_exp_t
+
+* Functions
+
+    * ${t}MetricProgrammableGetExp
+    * ${t}MetricProgrammableGetPropertiesExp
+    * ${t}MetricProgrammableGetParamDescriptionExp
+    * ${t}MetricProgrammableGetParamValueDescriptionExp
+    * ${t}MetricCreateFromProgrammableExp
+    * ${t}MetricGroupCreateExp
+    * ${t}MetricGroupAddMetricExp
+    * ${t}MetricDestroyExp
+    * ${t}MetricGroupDestroyExp
+
+Metric Programmable
+~~~~~~~~~~~~~~~~~~~
+
+Application can use ${t}MetricGroupGet to enumerate the list of metric groups and the metrics available for collection.
+However Applications may also want to collect custom metrics which are not listed as part of the enumeration.
+This is made possible by making a list of programmable metrics available to the application, so that it choose different parameter values for the preparing custom metrics.
```

Comment:
suggest reword "so that it may choose different parameter values for preparing custom metrics"

Reply:
Ok.

Reply:
fixed